### PR TITLE
test/docs: Improve docstrings and tests for utils inside of hex-strings.ts #2053

### LINKED ISSUE #2053: Improve docstrings and tests for utils inside of hex-strings.ts
This is part of a series of good first issues with the aim to improve the test coverage of the `core-utils` package. Our goal is to get to 100% test coverage within this package :-)

We need to improve the docstrings and test coverage of the utilities that can be found in the [hex-strings.ts](https://github.com/ethereum-optimism/optimism/blob/develop/packages/core-utils/src/common/hex-strings.ts) file within the `core-utils` package. Here is a [coverage report for this file](https://app.codecov.io/gh/ethereum-optimism/optimism/blob/develop/packages/core-utils/src/common/hex-strings.ts) which shows the lines that need to be covered by tests.

The following functions also need docstrings:
- [toRpcHexString](https://github.com/ethereum-optimism/optimism/blob/eb926e2f90b5eec4558f3c82746b639b19ddf9d5/packages/core-utils/src/common/hex-strings.ts#L59)
- [padHexString](https://github.com/ethereum-optimism/optimism/blob/eb926e2f90b5eec4558f3c82746b639b19ddf9d5/packages/core-utils/src/common/hex-strings.ts#L74)
- [encodeHex](https://github.com/ethereum-optimism/optimism/blob/eb926e2f90b5eec4558f3c82746b639b19ddf9d5/packages/core-utils/src/common/hex-strings.ts#L82)
- [hexStringEquals](https://github.com/ethereum-optimism/optimism/blob/eb926e2f90b5eec4558f3c82746b639b19ddf9d5/packages/core-utils/src/common/hex-strings.ts#L85)
- [bytes32ify](https://github.com/ethereum-optimism/optimism/blob/eb926e2f90b5eec4558f3c82746b639b19ddf9d5/packages/core-utils/src/common/hex-strings.ts#L97)

Please see [toHexString](https://github.com/ethereum-optimism/optimism/blob/eb926e2f90b5eec4558f3c82746b639b19ddf9d5/packages/core-utils/src/common/hex-strings.ts#L45-L51) for an example of a good docstring.

Feel free to leave a comment here if you would like to work on this issue and you are not sure about the purpose of these functions. Then I can explain the functions for you so that you'll be able to write a good docstring!

### PR DESCRIPTION
**Pull request by jgresham**
**Description**
Improve docstrings and tests for utils inside of hex-strings.ts #2053

**Additional context**
I wrote docstrings & tests to confirm the current functionality of hex-strings.ts. I do not have a good understanding of what could be wrong in the functions of hex-strings.ts or how they're used.

Notes:

1. I see duplicate code in packages/data-transport-layer/src/utils/common.ts
2. `padHexString` is confusing & should maybe add a `if (!ethers.utils.isHexString(inputString))` check

**Metadata**
- Fixes #2053

### CODE REVIEW

**Comment by jgresham** [comment]
@smartcontracts I did not create a changeset. I'm not sure if docstrings warrant a version bump.

File: packages/core-utils/src/common/hex-strings.ts [hunk 1]
```diff
@@ -79,9 +93,25 @@ export const padHexString = (str: string, length: number): string => {
...
+ * @return Hex string with '0' prefix
+ */
+export const encodeHex = (val: any, len: number): string =>
   remove0x(BigNumber.from(val).toHexString()).padStart(len, '0')
 
+/**
+ * Case insensitive hex string equality check
+ *
+ * @param stringA Hex string A
+ * @param  stringB Hex string B
```
Comment (smartcontracts) [inline packages/core-utils/src/common/hex-strings.ts:111]: Nit, extra space here
```suggestion
 * @param stringB Hex string B
```

File: packages/core-utils/test/hex-utils.spec.ts [hunk 2]
```diff
@@ -52,13 +56,20 @@ describe('add0x', () => {
 })
 
 describe('toHexString', () => {
-  it('should return undefined', () => {
-    expect(add0x(undefined)).to.deep.equal(undefined)
+  it('should throw an error when input is null', () => {
+    try {
```
Comment (smartcontracts) [inline packages/core-utils/test/hex-utils.spec.ts:60]: We generally try to stick to the following style when testing for errors:

```
expect(() => {
  toHexString(null)
}).to.throw('The first argument must be of type string or an instance of Buffer, ArrayBuffer, or Array or an Array-like Object. Received null')
```

**Review by smartcontracts — COMMENTED** [review]
Amazing work, thank you! Leaving some small comments, main thing here is around the assertions for errors. Otherwise LGTM!

**Comment by jgresham** [comment]
Made the changes above! Let me know if you want a rebase for any commits.

**Comment by smartcontracts** [comment]
@jgresham great work here, looks good to me! Would you mind doing the following quick steps:
1. Add a changeset so that we can get these docstrings released
2. Rebase this PR into a single commit using the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) format

Once that's done, this should be good to merge.

**Comment by jgresham** [comment]
After changset & rebase, other commits show up here. I'm not a rebase wizard, so was easier for me to create another PR https://github.com/ethereum-optimism/optimism/pull/2085

**Comment by smartcontracts** [comment]
> @smartcontracts After changset & rebase, other commits show up here. I'm not a rebase wizard, so was easier for me to create another PR #2085

Sounds good to me. I'll go ahead and close this one out.

**Pull request closed without merging**